### DISCUSSION TITLE
What Issues Does Uninstalling Distros Fix? Apt?

### DISCUSSION THREAD

**Discussion by kiot389** [microsoft/WSL#8720] [Q&A]
Last year I was troubleshooting a networking issue and found discussion that said uninstalling the distor fixes problems because it recompiles apt. The issue was related to Windows LAN Setting change. The company changed the proxy configuration with a new vendor. They started using a .pac script instead of a url:8080.  I reinstalled the distro and the problem was fixed. It was stated that WSL pulls down Windows settings and specifically configures apt; which is why it fixes a lot of issues.

Does reinstalling the distro fix issues because of the fundamental recompiling of apt?
I don't what to restate this when helping others troubleshoot.

**Comment by Biswa96** [comment]
Reinstalling the distro does **not** recompile or modify any program. It only removes the current distro in WSL and install a clear one same as provided in the root tarball.

**Reply by kiot389** [reply]
Does Windows' configuration affect the tarball in anyway? Do any files pull down settings from the host?

**Reply by kiot389** [reply]
For example, winhome is a variable setup.

**Reply by Biswa96** [reply]
The user distribution tarball is unaffected by any WSL steps. But after extraction/installation as WSL distribution, multiple things can happen e.g. /etc/resolv.conf is added etc.

Assume the tarball as a negative film in old cameras. The film is unaffected during development process but various things happen on the developing paper.

**Reply by kiot389** [reply]
Do any of the various things affect apt in anyway?

**Reply by Biswa96** [reply]
No, WSL does not modify any ELF binaries. You can verify that by using checksum.
